### ISSUE TITLE
Extraction of Vocals not 100% accurate

### ISSUE THREAD

**Issue by MagicAndy85** [boy1dr/SpleeterGui#56]
Dear Developers,

I've tested Spleeter with the Audio of this YouTube-Link:
[https://www.YouTube.com/watch?v=8nrdHcLM_ZA](https://www.YouTube.com/watch?v=8nrdHcLM_ZA)

It seems to Me, that the Vocals aren't extracted 100% accurate.

Maybe, You could improve Your Audio Extraction with this Sample. ;)

Yours, MagicAndy85

**Comment by Gusher** [comment]
Did you try other separation appa like RX8 or Phonicmind to compare results?

**Comment by boy1dr** [comment]
That track 2U is what we call "overcooked". The vocals are almost robotic thanks to autotune, the compression and DSP's at work there make the audio so mashed up i would be surprised if any source separation tool would work well with it.
Given a decent amount of post-processing work (like RX8 would do) you could maybe clean it up a bit but not much.
Spleeter would need to be trained and evaluated on such tracks to effectively separate it. The MusDB training set is far from that source audio though.
In any case, how would you teach an AI what the difference between a synthesizer and the vocals in that track are if they are almost indistinguishable?

**Comment by boy1dr** [comment]
for best results you should be working with audio recorded with lossless compression like PCM/WAV or FLAC.

**Comment by Poiuy69** [comment]
@boy1dr I get much better quality using demucs rather than spleeter. It's slower, but worth it. Maybe we need a demucs gui, hint hints lol😃 
https://github.com/facebookresearch/demucs

**Comment by boy1dr** [comment]
@Poiuy69 i have checked demucs out and i agree there are fewer artifacts in the resulting stems. and i did evaluate it for a GUI but it doesn't appear to be an "app" that can be downloaded and run like Spleeter can so for the time being i'll just keep an eye on it and see where it goes ;)

**Comment by iFUCKINGHATEcomputers** [comment]
> i did evaluate it for a GUI but it doesn't appear to be an "app" that can be downloaded and run like Spleeter

Interesting, does it depend on an external service or something?

**Comment by Gusher** [comment]
I hate to say but DEMUCS doesn't give significant results compared to Spleeter or RX8 or even online methods such as Phonicmind.... to get more accurate clean results you need the raw source of audio without any filters used on them, that's the key for very good output... alot of the music online and youtube are heavily filtered, making it useless for getting excellent results. However, I might rate RX8 a few points for adding extra tools for cleanup, such as de-bleed....

**Comment by Zcooger** [comment]
Have you tried MultiGenreModelHP.pth from UVR GUI? https://github.com/Anjok07/ultimatevocalremovergui/releases
![image](https://user-images.githubusercontent.com/42012614/95006235-b7922e80-05f1-11eb-8816-3938dd9a5b85.png)

**Comment by Gusher** [comment]
In its ccurrent state no way will touch "Ultimate Vocal Remover"... it needs a long way to compete against SpleeterGUI...

**Comment by boy1dr** [comment]
@iFUCKINGHATEcomputers i'm not sure how i'd use a colab project in a GUI. Just had a quick look at it again but the reason i wouldn't do a windows GUI for demucs is that it uses Anaconda python which is a separate install process. 
I do like making things but when a project gets as popular as SpleeterGUI you really don't want to have to provide a lot of support for failed installations.
If pip can be modified to install demucs and it works with the embeddable version of python it might be worth a shot :)

**Comment by Poiuy69** [comment]
@boy1dr I'm not sure what you mean by "app". Both deezer and demucs are just python libraries. Maybe I'm missing something

**Comment by boy1dr** [comment]
@Poiuy69 Apologies i hadn't looked back at demucs when i wrote that comment. it is just a python program but it uses Anaconda to install. If there is a PIP method i might look in to it but being Anaconda i'm not going to support a third party installation given the volume of people downloading SpleeterGUI i just wouldn't be able to manage it.

**Comment by AnshulJ999** [comment]
I just tried Demucs and came across this thread. I've been using SpleeterGUI, and it's a fantastic application. But after doing some very initial tests on Demucs, the results are definitely better. Especially for drums and bass. Much clearer. It seems like the 'demucs extra' library works well for less popular genres like progressive metal which usually have a lot of layers in different frequencies. But, demucs lacks clarity in the 'other' track, where it seems to be muddy and does not seem to have all the higher frequencies. 

A request to the developer of SpleeterGUI @boy1dr  - is it possible to include demucs into the application? As per your comments, there is a PIP method that I used to install demucs ( I have absolute no experience with coding whatsoever), and it was pretty easy to run. Using this with the GUI that you've built would make it accessible to everyone very easily and quickly too. 

I think it may be worth combining stems from demucs and spleeter together to see what combinations work well.
